src/chat/ChannelPreview.js

```diff
@@ -0,0 +1,103 @@
+import React, { useEffect, useState, useContext } from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components/macro'
+import { getUserById, getMessages } from '../utils/services'
+import SocketContext from '../SocketContext'
+import { setToStorage } from '../utils/storage'
+import { CHAT_START } from '../events'
+
+ChannelPreview.propTypes = {
+  channel: PropTypes.object.isRequired,
+  currentUser: PropTypes.object.isRequired,
+  setCurrentChannel: PropTypes.func.isRequired,
+}
+
+export default function ChannelPreview({
+  channel,
+  currentUser,
+  setCurrentChannel,
+}) {
+  const [chatPartner, setChatPartner] = useState({})
+  const [lastMessage, setLastMessage] = useState('')
+  const socket = useContext(SocketContext)
+
+  useEffect(() => {
+    const abortController = new AbortController()
+    const signal = abortController.signal
+    const chatPartnerId = channel.members.filter(
+      member => member !== currentUser._id
+    )[0]
```

Comment:
Muss hier nicht ein Komma stehen vor dem Array? Außerdem, Anmerkung von MDN: If you pass an empty array ([]), the props and state inside the effect will always have their initial values. While passing [] as the second argument is closer to the familiar componentDidMount and componentWillUnmount mental model, there are usually **better solutions** to avoid re-running effects too often. Also, don’t forget that React defers running useEffect until after the browser has painted, so doing extra work is less of a problem.